geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/visitor/QueryPlanFilterVisitor.scala

```diff
@@ -112,6 +114,28 @@ class QueryPlanFilterVisitor(sft: SimpleFeatureType) extends DuplicatingFilterVi
   }
 
   private def includeEquivalent(f: Filter): Boolean = f == Filter.INCLUDE || isFilterWholeWorld(f)
+
+  override def visit(expression: Expression, extraData: scala.Any): Expression = {
+    val visitedExpression = super.visit(expression, extraData)
+
+    visitedExpression match {
+            case exp: Expression =>
```

Comment:
is this match necessary? it will always be an expression, right?

Reply:
Hmm... I was thinking that the return from the super.visit could potentially be something other than an Expression... 

I'm not sure though...

Reply:
the method signature is `Expression` - could be null maybe...